Fixes part of #1095: Addition of config file

Sarthak2601 opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes part of #1095 
- Addition of the Firebase's `google-services.json` config file

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

Sarthak2601 committed:
config file

vinitamurthi reviewed (approved):
Wait on @BenHenning to approve

BenHenning commented on app/google-services.json:
```diff
@@ -0,0 +1,40 @@
+{
+  "project_info": {
+    "project_number": "409851798978",
```
How many of these values need to be real? Can we check in fake values for now, then fill in the correct values once we have a shared developer-only Firebase project to reference? I plan to create both a developer and production project to make this simpler.

Sarthak2601 replied:
From what I have seen, the package name should be the same and others can be changed. The only downside I see is that it will throw a lot of errors in the logcat when we use the dummy values, will that be okay ?

BenHenning replied:
Are these errors just at startup, or repeated? It's not ideal, and we should move to a debug project soon but some short-term errors just on app startup would be fine.

Can you file issues for creating the developer & production projects, and assign them to me? I defer to @vinitamurthi when these should be done by, but I'll make sure my part is done within whatever milestone we bucket them into.

Sarthak2601 replied:
They are repeated errors, not just at the startup. Okay, I'll create those issues.

vinitamurthi replied:
@BenHenning I think we should allow this PR to go in with the fake data because all of the other PRs that @Sarthak2601 is going to create for this milestone is dependent on this PR (the build fails without this json so there has to be a json in for any of the firebase work to begin). So either we get the development project with the proper data ready ASAP, or we let this PR go in and then you can get the development project ready by maybe 1 week before milestone 1 ends.

BenHenning replied:
@vinitamurthi will it be fine to get this checked in early next week? I can get the development project created next Monday to unblock this PR. The alternative seems to be pretty bad: repeated errors in logcat.

vinitamurthi replied:
Resolving this since we the new project has been created

aggarwalpulkit596 commented:
Can't we add a application suffix id for debug build ?

On Thu, Jun 4, 2020, 2:57 PM Sarthak Agarwal <notifications@github.com>
wrote:

> *@Sarthak2601* commented on this pull request.
> ------------------------------
>
> In app/google-services.json
> <https://github.com/oppia/oppia-android/pull/1241#discussion_r435116670>:
>
> > @@ -0,0 +1,40 @@
> +{
> +  "project_info": {
> +    "project_number": "409851798978",
>
> From what I have seen, the package name should be the same and others can
> be changed. The only downside I see is that it will throw a lot of errors
> in the logcat when we use the dummy values, will that be okay ?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia-android/pull/1241#discussion_r435116670>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG6KGSSPY3EKAM2DJZJ42P3RU5SGRANCNFSM4NRU3AQA>
> .
>

Sarthak2601 committed:
dummy data

Sarthak2601 commented:
> Can't we add a application suffix id for debug build ?
> […](#)

We can and we might as well but that will be done when we set up our own development project in some time.

vinitamurthi committed:
Ktlint upgrade lint fix (#1242)

* Fixed new ktlint errors

* Fixed data module lint errors with new ktlint

* Add memory param to JAVA_TOOL_OPTIONS

* Add gradle options as well

* Remove java_tool_options

* Reduce test threads and increase ram of docker

* Fix docker image issue

* Removed large resource class since its not available for our plan

* Remove test fork config from gradle files and added it to the test command

* Attempt to build the app before running tests

* Build the app without any daemon/worker restrictions

* Test a new setup job on circleci

* Change indentation of config

* Fix setup

* Fix ktlint installation

* Revert android dependencies

* Fix unterminated string issue

* Add workspace attachment

* Fix workspace path

* Fix setup path

* Separate linters from the android dependencies

* Fix cache restore issue

* Split non flaky tests into module wise test jobs

* Set heap size in app tests

* Fixed wrong repo

* Testing changing java version'

* Add the command instead of an orb

* Update java home path

* Fix java home path

* Add java 11 installation to every job for verification

* Remove JDK 11 and enable build caching

* Reduce size for each daemon

* Remove splash tests from the non flaky app tests

vinitamurthi commented on app/google-services.json:
```diff
@@ -0,0 +1,40 @@
+  "client": [
+    {
+      "client_info": {
+        "mobilesdk_app_id": "1:123456789012:android:d544ded26ae57f7fe6e25b",
```
I think we should make it obvious that this is fake data..something like "1:1234...:android:fakestring" or something
From this it doesn't look very obvious

Sarthak2601 replied:
Done

MohamedMedhat1998 committed:
Fixes #1084: Tablet: Highfi Onboarding Flow (landscape) (#1115)

* -Onboarding: draft implementation for tablet-landscape high-fi

* -Removed the unused drawable file
-Used `app:layout_constraintDimensionRatio="H, 10:9"` and `android:scaleType="fitXY"` instead of `guideline` and `android:scaleType="centerCrop"`
-Centered the `textViews` between the `imageView` and the end of the `parentLayout` and gave them a fixed width of `360dp` instead of placing the `textViews` relative to the end of the `parentLayout` only

* -placed the textViews and the buttons relative to the end of the screen (with a margin of 96dp) and a guideline of 57% from the beginning of the screen instead of using fixed width of 360dp

* -Deleted "onboarding_dot_active.xml" with "sw600dp-land" qualifier
-Removed the unnecessary "FrameLayout" in "onboarding_fragment.xml"
-Removed the "guidelines" in "onboarding_slide.xml" and "onboarding_slide_final.xml" and replaced them with margins and fixed widths
-Updated the "oppiaOnboardingDivider" in "colors.xml" and used it in the "onboarding_fragment.xml"

* -Updated the UI

* -Added EOF in "booleans.xml"

* -Updated the ratio of the thumbnail images to be "40:39"

* -Changed the height of the "Get Started" button to be "wrap_content" (minHeight is set to 48dp)

aggarwalpulkit596 committed:
Fixes #1216 : Deleted Extra Proto Models (#1232)

* Remove Extra Proto Models

* try gradle changes

* try circle ci changes

* remove testing changes[skip CI]

aggarwalpulkit596 committed:
Fixes #1064 : Added DropDropSortInput Module (#1092)

* Add DropDropSortInput Module along with HasElementXwithPositionY Classifier

aggarwalpulkit596 committed:
Fixes #1221 : Added HasElementXBeforeElementY Classifier Rule [Blocked: #1092] (#1231)

* Add DragDropSortInputHasElementXBeforeElementYClassifierProvider

BenHenning commented on app/google-services.json:
```diff
@@ -0,0 +1,40 @@
+      },
+      "oauth_client": [
+        {
+          "client_id": "123456789012-4unv88bb0rj9c4ki29hanl213qg6ter8.apps.googleusercontent.com",
```
Ditto here--can we clearly indicate that this is a fake string?

Sarthak2601 replied:
Done

BenHenning commented on app/google-services.json:
```diff
@@ -0,0 +1,40 @@
+      ],
+      "api_key": [
+        {
+          "current_key": "BJabTzEHBlKeXx1u8fMxi58ZC5E-mwcZTB4nPi4"
```
And for this key.

Sarthak2601 replied:
Done

Sarthak2601 committed:
fake string

vinitamurthi commented:
Looks good! You probably want to pull from develop once to get the new CircleCI changes

Sarthak2601 committed:
dummy

Sarthak2601 committed:
Merge branch 'develop' into firebase-config-file

BenHenning commented:
@Sarthak2601 the developer project is now available. Please update the JSON file to match that & send back to me once that's done.

Sarthak2601 committed:
dev project config file

Sarthak2601 committed:
nit

Sarthak2601 commented:
> @Sarthak2601 the developer project is now available. Please update the JSON file to match that & send back to me once that's done.

JSON file updated, PTAL.

BenHenning reviewed (approved):
Thanks @Sarthak2601!
